File: BeautyTimeFrontEnd/src/service/cadastros/CidadeService.js
import axios from 'axios';

import Axios from "axios";
import {ServiceBase} from "./ServiceBase";

export class CidadeService {

    url = process.env.REACT_APP_URI_API + "cidade";

    listarTodos() {
        return Axios.get(this.url);
    }

    inserir(objeto) {
        return Axios.post(this.url, objeto);
    }

    alterar(objeto) {
        return Axios.put(this.url, objeto);
    }

    excluir(id) {
        return Axios.delete(this.url + id);
    }
}

File: BeautyTimeBackEnd/src/main/java/com/gustavoorelio/beautytime/service/CidadeService.java
package com.gustavoorelio.beautytime.service;

import com.gustavoorelio.beautytime.model.Cidade;
import com.gustavoorelio.beautytime.repository.CidadeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CidadeService {

    @Autowired
    private CidadeRepository cidadeRepository;

    public List<Cidade> buscarTodos() {
        return cidadeRepository.findAll();
    }

    public Cidade inserir(Cidade cidade) {
        Cidade cidadeNovo = cidadeRepository.saveAndFlush(cidade);
        return cidadeNovo;
    }

    public Cidade alterar(Cidade cidade) {
        return cidadeRepository.saveAndFlush(cidade);
    }

    public void excluir(Long id) {
        Cidade cidade = cidadeRepository.findById(id).get();
        cidadeRepository.delete(cidade);
    }
}
